micrometer-core/src/main/java/io/micrometer/core/aop/CountedAspect.java

```diff
@@ -198,17 +198,23 @@ public Object countedClass(ProceedingJoinPoint pjp) throws Throwable {
      * {@link Counted#recordFailuresOnly()} is set to {@code false}, a success is
      * recorded.
      * @param pjp Encapsulates some information about the intercepted area.
-     * @param counted The annotation.
      * @return Whatever the intercepted method returns.
      * @throws Throwable When the intercepted method throws one.
      */
-    @Around(value = "@annotation(counted)", argNames = "pjp,counted")
+    @Around("execution (@io.micrometer.core.annotation.Counted * *.*(..))")
     @Nullable
-    public Object interceptAndRecord(ProceedingJoinPoint pjp, Counted counted) throws Throwable {
+    public Object interceptAndRecord(ProceedingJoinPoint pjp) throws Throwable {
```

Comment:
Alexander kindly provided a fix for this that doesn't need to change the method signature https://github.com/micrometer-metrics/micrometer/issues/1149#issuecomment-2104899044

Reply:
Fixed :tada: no more binary incompatibility issues